### ISSUE TITLE
Dynmap leaves, podzol and grass not rendering properly

### ISSUE THREAD

**Issue by HugoNordenswan** [webbukkit/dynmap#3046]
![2020-07-28 10-10-25 00_00_00_00 Still001](https://user-images.githubusercontent.com/66118195/88638188-70506480-d0bb-11ea-8ec9-23dbc14fef50.png)

Hey! I've made a world in Worldpainter. Then I downloaded dynmap to my server and my map looks very strange. Leaves are invisible, podzol looks like snow and grass looks snowy on the sides. I've had this problem for a long time now and it's very annoying. Please, can anyone help?

This is supposed to look like a taiga biome. My server is 1.14.4 and I have the latest dynmap plugin for that version. Free rank for anyone that fixes this!

**Comment by FedUpWith-Tech** [comment]
What is the output of /dynmap version

**Comment by HugoNordenswan** [comment]
> What is the output of /dynmap version

Dynmap version: core=3.0-beta-5-218, plugin=3.0-beta-5-218

**Comment by FedUpWith-Tech** [comment]
Please use the [most recent version](https://github.com/webbukkit/dynmap/releases/tag/v3.1-beta-2) and re-render.

**Comment by HugoNordenswan** [comment]
> Please use the [most recent version](https://github.com/webbukkit/dynmap/releases/tag/v3.1-beta-2) and re-render.

But my server is 1.14.4, should I still use the newest dynmap version?

**Comment by FedUpWith-Tech** [comment]
Yes

**Comment by HugoNordenswan** [comment]
> Yes

Ok

**Comment by HugoNordenswan** [comment]
> Yes

Can I save the dynmap folder?

**Comment by FedUpWith-Tech** [comment]
> > Yes
> 
> Can I save the dynmap folder?

What do you mean?

**Comment by HugoNordenswan** [comment]
> > > Yes
> > 
> > 
> > Can I save the dynmap folder?
> 
> What do you mean?

The folder with the config file for example. I'll delete it.

**Comment by FedUpWith-Tech** [comment]
you only need to delete the old dynmap jar file and update. The do a /dynmap fullrender again

**Comment by HugoNordenswan** [comment]
> you only need to delete the old dynmap jar file and update. The do a /dynmap fullrender again

Ok

**Comment by HugoNordenswan** [comment]
> you only need to delete the old dynmap jar file and update. The do a /dynmap fullrender again

Rendering...

**Comment by HugoNordenswan** [comment]
> you only need to delete the old dynmap jar file and update. The do a /dynmap fullrender again

Nothing has changed.

**Comment by FedUpWith-Tech** [comment]
Can you send me a link to your map? 
Are you using custom texture packs or mods that would change leaves?

**Comment by HugoNordenswan** [comment]
> Can you send me a link to your map?
> Are you using custom texture packs or mods that would change leaves?

Of course http://51.89.100.195:25575/ No mods, no texturepacks.

**Comment by FedUpWith-Tech** [comment]
Are all the leaves that are missing spruce? It seems to have rendered some of the leaves/foliage. Can you pastebin configuration.txt

**Comment by HugoNordenswan** [comment]
> Are all the leaves that are missing spruce? It seems to have rendered some of the leaves/foliage. Can you pastebin configuration.txt

Not all missing leaves are spruce. Most acacia and jungle leaves are also missing. https://pastebin.com/LX4mMnr5

**Comment by FedUpWith-Tech** [comment]
What version of minecraft was this map created on?

**Comment by HugoNordenswan** [comment]
> What version of minecraft was this map created on?

1.14.4 I still have the worldpainter file.

**Comment by HugoNordenswan** [comment]
> What version of minecraft was this map created on?

This bug seems to only happen when I use a worldpainter world.

**Comment by FedUpWith-Tech** [comment]
try uncommenting out this line (last in the config) `#migrate-chunks: true` and do a /dynmap fullrender again

**Comment by HugoNordenswan** [comment]
> try uncommenting out this line (last in the config) `#migrate-chunks: true` and do a /dynmap fullrender again

What is uncommenting? Changing to false?

**Comment by FedUpWith-Tech** [comment]
change `#migrate-chunks: true` to this `migrate-chunks: true`

**Comment by HugoNordenswan** [comment]
> change `#migrate-chunks: true` to this `migrate-chunks: true`

Okay

**Comment by HugoNordenswan** [comment]
> change `#migrate-chunks: true` to this `migrate-chunks: true`

The map is rendered. But many leaves are still missing. http://51.89.100.195:25575/?worldname=Survival%20World%20by%20Hugo%20Nordenswan&mapname=flat&zoom=2&x=168&y=64&z=709#

**Comment by FedUpWith-Tech** [comment]
did you do a /dynmap fullrender?

**Comment by HugoNordenswan** [comment]
Yep

**Comment by HugoNordenswan** [comment]
Oh, it was still running...

**Comment by HugoNordenswan** [comment]
> did you do a /dynmap fullrender?

Now I have rendered multiple times and t

**Comment by HugoNordenswan** [comment]
> did you do a /dynmap fullrender?

Now I have rendered my map multiple times and many leaves still are not rendered.

**Comment by HugoNordenswan** [comment]
I am also getting this in the console.

[INFO] [dynmap] Unknown Legacy block ID in CTM: 1

**Comment by FedUpWith-Tech** [comment]
Did you purge your map before rendering?

**Comment by bloggy** [comment]
Same issue here. Really need a fix as more and more people use worldpainter maps along with dynmap only to find the leaves not working / displayed on the map.

**Comment by Mint-101** [comment]
I need a fix for this. A lot of us have this bug and nobody's done anything about it. Is there anything I can do to fix this? I did everything the post did and none of it worked.

**Comment by Mint-101** [comment]
I actually found a way to fix this. It's not perfect but if you load in some chunks on your Minecraft client Dynmap will fix the leaves. Eventually, when your map has been fully loaded by players there will be no more invisible leaves. It's a scuffed solution but for many, it's just going to have to do.

EDIT: I was, unfortunately, wrong but I did learn something else. Some of my Dynmap custom trees rendered correctly although most didn't. I do not have the time to go more in-depth with this and test different custom trees to see which ones render, but it seems to me that trees with matching log and leaf types will render correctly.
